[quant] Re-enable test_nested that has different qconfig for shared ClassType

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [WIP][quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32543 [WIP][quant][graphmode] ConvBn Folding works with bias=None
* #32380 [quant][graphmode][refactor] Better API for fold_convbn
* #32374 [refactor] Adding FoldConvBatchNorm2dHelper
* #32303 [quant][graphmode] Call _jit_pass_dedup_module_ueses in quantize_script
* #32302 [quant] Re-enable fold_convbn in quantize_script
* **#32206 [quant] Re-enable test_nested that has different qconfig for shared ClassType**
* #32205 [quant][graphmode] Support quantizing shared ClassType with different qconfigs

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508028](https://our.internmc.facebook.com/intern/diff/D19508028)

jerryzh168 committed:
[quant] Re-enable test_nested that has different qconfig for shared ClassType

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 6d080450:


* **8/8** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=6d08045081af0ebd68f5e51df8a72e1ff2a3de7f).

### :detective: 8 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248331) pytorch_xla_linux_xenial_py3_6_clang7_build (1/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86222775) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86222775))
<details>
<summary>
<code>Jan 15 02:40:07 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 15 02:40:07 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 15 02:40:07 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:40:07 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:40:07 1 out of 1 hunk FAILED 
Jan 15 02:40:07 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:40:07 Traceback (most recent call last): 
Jan 15 02:40:07   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 15 02:40:07     patch_repo(args) 
Jan 15 02:40:07   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 15 02:40:07     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 15 02:40:07 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:40:07 + cleanup 
Jan 15 02:40:07 + retcode=1 
Jan 15 02:40:07 + set +x 
Jan 15 02:40:07 =================== sccache compilation log =================== 
Jan 15 02:40:07 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 15 02:40:07 Compile requests                 1 
Jan 15 02:40:07 Compile requests executed        0 
Jan 15 02:40:07 Cache hits                       0 
Jan 15 02:40:07 Cache misses                     0 
Jan 15 02:40:07 Cache timeouts                   0 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248387) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (2/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86225575) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86225575))
<details>
<summary>
<code>Jan 15 02:43:54 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:890:33: error: call to function 'operator==' that is neither visible in the template definition nor found by argument-dependent lookup</code>
</summary>

```
Jan 15 02:43:48 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/schema_matching.cpp.o 
Jan 15 02:43:49 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/script_type_parser.cpp.o 
Jan 15 02:43:54 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.cpp:1: 
Jan 15 02:43:54 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.h:8: 
Jan 15 02:43:54 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/ir.h:3: 
Jan 15 02:43:54 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/attributes.h:2: 
Jan 15 02:43:54 In file included from /var/lib/jenkins/workspace/aten/src/ATen/ATen.h:3: 
Jan 15 02:43:54 In file included from /var/lib/jenkins/workspace/c10/core/Allocator.h:4: 
Jan 15 02:43:54 In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/memory:79: 
Jan 15 02:43:54 In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/functional:55: 
Jan 15 02:43:54 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:890:33: error: call to function 'operator==' that is neither visible in the template definition nor found by argument-dependent lookup 
Jan 15 02:43:54         return bool(std::get<__i>(__t) == std::get<__i>(__u)) 
Jan 15 02:43:54                                        ^ 
Jan 15 02:43:54 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:923:25: note: in instantiation of member function 'std::__tuple_compare<std::tuple<torch::jit::script::Module, torch::jit::script::Module>, std::tuple<torch::jit::script::Module, torch::jit::script::Module>, 0, 2>::__eq' requested here 
Jan 15 02:43:54       return __compare::__eq(__t, __u); 
Jan 15 02:43:54                         ^ 
Jan 15 02:43:54 /var/lib/jenkins/workspace/c10/util/Optional.h:633:68: note: in instantiation of function template specialization 'std::operator==<torch::jit::script::Module, torch::jit::script::Module, torch::jit::script::Module, torch::jit::script::Module>' requested here 
Jan 15 02:43:54   return bool(x) != bool(y) ? false : bool(x) == false ? true : *x == *y; 
Jan 15 02:43:54                                                                    ^ 
Jan 15 02:43:54 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/bits/stl_function.h:357:20: note: in instantiation of function template specialization 'c10::operator==<std::tuple<torch::jit::script::Module, torch::jit::script::Module> >' requested here 
Jan 15 02:43:54       { return __x == __y; } 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248357) pytorch_linux_xenial_py3_clang5_asan_build (3/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86227731) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86227731))
<details>
<summary>
<code>Jan 15 02:45:02 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:890:33: error: call to function 'operator==' that is neither visible in the template definition nor found by argument-dependent lookup</code>
</summary>

```
Jan 15 02:44:57 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/module.cpp.o 
Jan 15 02:44:57 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/object.cpp.o 
Jan 15 02:45:02 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.cpp:1: 
Jan 15 02:45:02 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.h:8: 
Jan 15 02:45:02 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/ir.h:3: 
Jan 15 02:45:02 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/attributes.h:2: 
Jan 15 02:45:02 In file included from /var/lib/jenkins/workspace/aten/src/ATen/ATen.h:3: 
Jan 15 02:45:02 In file included from /var/lib/jenkins/workspace/c10/core/Allocator.h:4: 
Jan 15 02:45:02 In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/memory:79: 
Jan 15 02:45:02 In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/functional:55: 
Jan 15 02:45:02 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:890:33: error: call to function 'operator==' that is neither visible in the template definition nor found by argument-dependent lookup 
Jan 15 02:45:02         return bool(std::get<__i>(__t) == std::get<__i>(__u)) 
Jan 15 02:45:02                                        ^ 
Jan 15 02:45:02 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:923:25: note: in instantiation of member function 'std::__tuple_compare<std::tuple<torch::jit::script::Module, torch::jit::script::Module>, std::tuple<torch::jit::script::Module, torch::jit::script::Module>, 0, 2>::__eq' requested here 
Jan 15 02:45:02       return __compare::__eq(__t, __u); 
Jan 15 02:45:02                         ^ 
Jan 15 02:45:02 /var/lib/jenkins/workspace/c10/util/Optional.h:633:68: note: in instantiation of function template specialization 'std::operator==<torch::jit::script::Module, torch::jit::script::Module, torch::jit::script::Module, torch::jit::script::Module>' requested here 
Jan 15 02:45:02   return bool(x) != bool(y) ? false : bool(x) == false ? true : *x == *y; 
Jan 15 02:45:02                                                                    ^ 
Jan 15 02:45:02 /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/bits/stl_function.h:357:20: note: in instantiation of function template specialization 'c10::operator==<std::tuple<torch::jit::script::Module, torch::jit::script::Module> >' requested here 
Jan 15 02:45:02       { return __x == __y; } 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248332) pytorch_macos_10_13_py3_build (4/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86227817) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86227817))
<details>
<summary>
<code>Jan 14 18:45:43 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o</code>
</summary>

```
Jan 14 18:45:21             ~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~ 
Jan 14 18:45:21 1 warning generated. 
Jan 14 18:45:22 [2491/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/requires_grad_analysis.cpp.o 
Jan 14 18:45:22 [2492/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/specialize_autogradzero.cpp.o 
Jan 14 18:45:31 [2493/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/subgraph_rewrite.cpp.o 
Jan 14 18:45:32 [2494/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/subgraph_utils.cpp.o 
Jan 14 18:45:32 [2495/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/check_alias_annotation.cpp.o 
Jan 14 18:45:32 [2496/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/memory_dag.cpp.o 
Jan 14 18:45:35 [2497/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/python_print.cpp.o 
Jan 14 18:45:43 [2498/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o 
Jan 14 18:45:43 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o  
READ -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -c ../torch/csrc/jit/passes/quantization.cpp 
Jan 14 18:45:43 In file included from ../torch/csrc/jit/passes/quantization.cpp:1: 
Jan 14 18:45:43 In file included from ../torch/csrc/jit/passes/quantization.h:8: 
Jan 14 18:45:43 In file included from ../torch/csrc/jit/ir.h:3: 
Jan 14 18:45:43 In file included from ../torch/csrc/jit/attributes.h:2: 
Jan 14 18:45:43 In file included from ../aten/src/ATen/ATen.h:3: 
Jan 14 18:45:43 In file included from ../c10/core/Allocator.h:4: 
Jan 14 18:45:43 In file included from /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:653: 
Jan 14 18:45:43 /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/tuple:1126:77: error: invalid operands to binary expression ('const typename tuple_element<0UL, tuple<Module, Module> >::type' (aka 'const __type_pack_element<0UL, torch::jit::script::Module, torch::jit::script::Module>') and 'const typename tuple_element<0UL, tuple<Module, Module> >::type') 
Jan 14 18:45:43         return __tuple_equal<_Ip - 1>()(__x, __y) && _VSTD::get<_Ip-1>(__x) == _VSTD::get<_Ip-1>(__y); 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248330) pytorch_macos_10_13_cuda10_0_cudnn7_py3_build (5/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86229428) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86229428))
<details>
<summary>
<code>Jan 14 18:48:39 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o</code>
</summary>

```
Jan 14 18:48:37 [2517/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/string_to_type.cpp.o 
Jan 14 18:48:37 [2518/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/source_range_serialization.cpp.o 
Jan 14 18:48:37 [2519/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/tracer.cpp.o 
Jan 14 18:48:38 [2520/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/hooks_for_testing.cpp.o 
Jan 14 18:48:38 [2521/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/utils/tensor_flatten.cpp.o 
Jan 14 18:48:38 [2522/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/utils/variadic.cpp.o 
Jan 14 18:48:39 [2523/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/fuser/kernel_cache.cpp.o 
Jan 14 18:48:39 [2524/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/fuser/compiler.cpp.o 
Jan 14 18:48:39 [2525/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/fuser/executor.cpp.o 
Jan 14 18:48:39 [2526/2988] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o 
Jan 14 18:48:39 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o  
READ -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -c ../torch/csrc/jit/passes/quantization.cpp 
Jan 14 18:48:39 In file included from ../torch/csrc/jit/passes/quantization.cpp:1: 
Jan 14 18:48:39 In file included from ../torch/csrc/jit/passes/quantization.h:8: 
Jan 14 18:48:39 In file included from ../torch/csrc/jit/ir.h:3: 
Jan 14 18:48:39 In file included from ../torch/csrc/jit/attributes.h:2: 
Jan 14 18:48:39 In file included from ../aten/src/ATen/ATen.h:3: 
Jan 14 18:48:39 In file included from ../c10/core/Allocator.h:4: 
Jan 14 18:48:39 In file included from /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:653: 
Jan 14 18:48:39 /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/tuple:1126:77: error: invalid operands to binary expression ('const typename tuple_element<0UL, tuple<Module, Module> >::type' (aka 'const __type_pack_element<0UL, torch::jit::script::Module, torch::jit::script::Module>') and 'const typename tuple_element<0UL, tuple<Module, Module> >::type') 
Jan 14 18:48:39         return __tuple_equal<_Ip - 1>()(__x, __y) && _VSTD::get<_Ip-1>(__x) == _VSTD::get<_Ip-1>(__y); 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248368) pytorch_linux_xenial_py3_clang5_android_ndk_r19c_mobile_code_analysis (6/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86235927) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86235927))
<details>
<summary>
<code>Jan 15 02:59:13 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o</code>
</summary>

```
Jan 15 02:58:59         if (*type->dim() >= num_reduced_dim && num_reduced_dim > 0) { 
Jan 15 02:58:59             ~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~ 
Jan 15 02:58:59 1 warning generated. 
Jan 15 02:59:00 [566/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/subgraph_rewrite.cpp.o 
Jan 15 02:59:02 [567/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/memory_dag.cpp.o 
Jan 15 02:59:08 [568/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/subgraph_utils.cpp.o 
Jan 15 02:59:11 [569/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/check_alias_annotation.cpp.o 
Jan 15 02:59:12 [570/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/print_handler.cpp.o 
Jan 15 02:59:12 [571/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/python_print.cpp.o 
Jan 15 02:59:13 [572/613] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o 
Jan 15 02:59:13 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o  
rameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -Wno-maybe-uninitialized -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -c /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.cpp 
Jan 15 02:59:13 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.cpp:1: 
Jan 15 02:59:13 In file included from ../../torch/csrc/jit/passes/quantization.h:8: 
Jan 15 02:59:13 In file included from ../../torch/csrc/jit/ir.h:3: 
Jan 15 02:59:13 In file included from ../../torch/csrc/jit/attributes.h:2: 
Jan 15 02:59:13 In file included from ../../aten/src/ATen/ATen.h:3: 
Jan 15 02:59:13 In file included from ../../c10/core/Allocator.h:4: 
Jan 15 02:59:13 In file included from /usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/memory:79: 
Jan 15 02:59:13 In file included from /usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/functional:55: 
Jan 15 02:59:13 /usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:890:33: error: call to function 'operator==' that is neither visible in the template definition nor found by argument-dependent lookup 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248377) pytorch_linux_xenial_py3_clang5_android_ndk_r19c_x86_32_build (7/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86236165) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86236165))
<details>
<summary>
<code>Jan 15 03:00:03 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o</code>
</summary>

```
Jan 15 02:59:41 [533/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/requires_grad_analysis.cpp.o 
Jan 15 02:59:46 [534/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/shape_analysis.cpp.o 
Jan 15 02:59:46 [535/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/specialize_autogradzero.cpp.o 
Jan 15 02:59:51 [536/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/subgraph_rewrite.cpp.o 
Jan 15 02:59:54 [537/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/subgraph_utils.cpp.o 
Jan 15 02:59:57 [538/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/python_print.cpp.o 
Jan 15 02:59:57 [539/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/memory_dag.cpp.o 
Jan 15 03:00:02 [540/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/check_alias_annotation.cpp.o 
Jan 15 03:00:02 [541/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/print_handler.cpp.o 
Jan 15 03:00:03 [542/583] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o 
Jan 15 03:00:03 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o  
rameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -Wno-maybe-uninitialized -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -c /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.cpp 
Jan 15 03:00:03 In file included from /var/lib/jenkins/workspace/torch/csrc/jit/passes/quantization.cpp:1: 
Jan 15 03:00:03 In file included from ../torch/csrc/jit/passes/quantization.h:8: 
Jan 15 03:00:03 In file included from ../torch/csrc/jit/ir.h:3: 
Jan 15 03:00:03 In file included from ../torch/csrc/jit/attributes.h:2: 
Jan 15 03:00:03 In file included from ../aten/src/ATen/ATen.h:3: 
Jan 15 03:00:03 In file included from ../c10/core/Allocator.h:4: 
Jan 15 03:00:03 In file included from /opt/ndk/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:663: 
Jan 15 03:00:03 /opt/ndk/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/tuple:1122:77: error: invalid operands to binary expression ('const typename tuple_element<0U, tuple<Module, Module> >::type' (aka 'const __type_pack_element<0U, torch::jit::script::Module, torch::jit::script::Module>') and 'const typename tuple_element<0U, tuple<Module, Module> >::type') 
Jan 15 03:00:03         return __tuple_equal<_Ip - 1>()(__x, __y) && _VSTD::get<_Ip-1>(__x) == _VSTD::get<_Ip-1>(__y); 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248323) binary_macos_libtorch_2_7_cpu_build (8/8)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86259614) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86259614))
<details>
<summary>
<code>Jan 15 03:33:21 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o</code>
</summary>

```
Jan 15 03:33:08 ../torch/csrc/jit/passes/shape_analysis.cpp:1244:26: warning: comparison of integers of different signs: 'unsigned long' and 'int64_t' (aka 'long long') [-Wsign-compare] 
Jan 15 03:33:08         if (*type->dim() >= num_reduced_dim && num_reduced_dim > 0) { 
Jan 15 03:33:08             ~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~ 
Jan 15 03:33:08 1 warning generated. 
Jan 15 03:33:09 [2505/3005] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/memory_dag.cpp.o 
Jan 15 03:33:13 [2506/3005] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/subgraph_utils.cpp.o 
Jan 15 03:33:17 [2507/3005] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/python_print.cpp.o 
Jan 15 03:33:17 [2508/3005] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/utils/check_alias_annotation.cpp.o 
Jan 15 03:33:17 [2509/3005] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/print_handler.cpp.o 
Jan 15 03:33:21 [2510/3005] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o 
Jan 15 03:33:21 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o  
READ -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/quantization.cpp.o -c ../torch/csrc/jit/passes/quantization.cpp 
Jan 15 03:33:21 In file included from ../torch/csrc/jit/passes/quantization.cpp:1: 
Jan 15 03:33:21 In file included from ../torch/csrc/jit/passes/quantization.h:8: 
Jan 15 03:33:21 In file included from ../torch/csrc/jit/ir.h:3: 
Jan 15 03:33:21 In file included from ../torch/csrc/jit/attributes.h:2: 
Jan 15 03:33:21 In file included from ../aten/src/ATen/ATen.h:3: 
Jan 15 03:33:21 In file included from ../c10/core/Allocator.h:4: 
Jan 15 03:33:21 In file included from /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:653: 
Jan 15 03:33:21 /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/tuple:1126:77: error: invalid operands to binary expression ('const typename tuple_element<0UL, tuple<Module, Module> >::type' (aka 'const __type_pack_element<0UL, torch::jit::script::Module, torch::jit::script::Module>') and 'const typename tuple_element<0UL, tuple<Module, Module> >::type') 
Jan 15 03:33:21         return __tuple_equal<_Ip - 1>()(__x, __y) && _VSTD::get<_Ip-1>(__x) == _VSTD::get<_Ip-1>(__y); 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
👍

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Re-enable test_nested that has different qconfig for shared ClassType"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508028](https://our.internmc.facebook.com/intern/diff/D19508028)

[ghstack-poisoned]
